### ISSUE TITLE
Implement modal for Generic Proposal

### ISSUE THREAD

**Issue by taoeffect** [okTurtles/group-income#1303]
### Problem

The implementation for the new generic proposal design (#867) doesn't exist.

### Solution

Once #1300 is closed, implement generic proposals in the contracts and the modal for it.

**Comment by SebinSong** [comment]
- "create proposals" button design figma link:
https://github.com/okTurtles/group-income/issues/1288#issuecomment-1155061508
- "modal" design figma link:
https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=23051%3A213049

**Comment by SebinSong** [comment]
@taoeffect 
Things that I've completed working on as part of this task atm are,

1. building the Generic Proposal Modal(`GenericProposal.vue`). 
![image](https://user-images.githubusercontent.com/17641213/179202815-7f698a22-17a5-45a9-940c-96a24b482483.png)

2. make sure the 'open' generic proposal item shows up in the proposal widget in the Dashboard.
![image](https://user-images.githubusercontent.com/17641213/179204266-2117fba6-f546-4497-84bb-464591eaed2d.png)

Do you think these two cover the scope of this task item and deserve a PR to master? (I didn't touch any contract file, as no idea what to do with them yet..!)

**Comment by SebinSong** [comment]
@taoeffect 
I just sent a [PR here](https://github.com/okTurtles/group-income/pull/1353) for my work for this item.
I noticed there's no call-to-action element implemented yet in the app that triggers a generic proposal modal, so
created this **temporary** 'Create general proposal' button in `ProposalWidget.vue` in the dashboard.

![image](https://user-images.githubusercontent.com/17641213/179385631-0730eebc-bb86-49aa-bc4a-3cb2928a3b30.png)

clicking on it will pop out the modal that I've implemented as screenshot below.

![image](https://user-images.githubusercontent.com/17641213/179385689-26ffb13d-31df-4b7f-b218-7077ae8f0739.png)

Looking forwards to your review and change requests for further improvements in the work.
